webdiff/app.py

```diff
@@ -279,11 +282,19 @@ def run():
 
     PORT = pick_a_port(parsed_args)
 
-    sys.stderr.write('''Serving diffs on http://localhost:%s
+    if app.config.get('USE_HOSTNAME'):
+        _hostname = platform.node()
+        # platform.node will return empty string if it can't find the hostname
+        if not _hostname:
+            sys.stderr.write('Warning: hostname could not be determined')
+        else:
+            HOSTNAME = _hostname
+
+    sys.stderr.write('''Serving diffs on http://%s:%s
 Close the browser tab or hit Ctrl-C when you're done.
-''' % PORT)
+''' % (HOSTNAME, PORT))
     Timer(0.1, open_browser).start()
-    app.run(host='0.0.0.0', port=PORT)
+    app.run(host=HOSTNAME, port=PORT)
```

Comment:
> But what I don't understand is, if my hostname is foo, then what's the difference between app.run(host='0.0.0.0', port=1234) and app.run(host='foo', port=1234)?

I think you're correct in thinking that there isn't a practical difference, in the case above. Like I mentioned previously I just changed it so that it's clear and the same everywhere (in code) what we're using for the webdiff flask server (localhost or client hostname) rather than have most places in the code change, but this line having a static/hardcoded value (which is somewhat unsecure in the case when you think you're just using webdiff locally, like you mentioned previously).

However, I also believe in "don't fix it if it ain't broke" :p So if you want me to unwind that bit back to ```run``` always using ```'0.0.0.0'``` I'd be happy to do so.